dev-docs/RFCs/v5.0/viewport-animation-rfc.md

```diff
@@ -60,11 +60,23 @@ viewportAnimationDuration: (Number) Animation duration in milliseconds, default
+3. **animationFunction** {Function, default: `viewportLinearAnimation`} : Function that gets called for each animation step to calculated animated viewport. It takes start, end viewports and animation step, returns animated viewport. We provide couple of utility functions, `viewportLinearAnimation` and `viewportFlyToAnimation`. By default `viewportLinearAnimation` is used where all viewport props are linearly animated. `viewportFlyToAnimation` animates viewports similar to MapBox `flyTo` API, this is pretty useful when camera center changes. But a user can provide any function for this prop to perform custom viewport animations.
+
+4. **animationFreeze** {bool, optional} : When set to `true` and if `animationDuration` not equal to 0, any following viewport updates are ignored until animation resulting this update is finished. This prop has no effect if `animationDuration` is 0. When it is set to `false` or not set, any following viewport updates will stop the current animation and update viewport according to other props.
+
+5. **animationCompleteOnUpdate** {bool, false} : This prop can be used to control how to end current animation, when new animation requested before it is completed, following table describes in more details.
+
+| Current viewport animating | Current Viewport start value| Current Viewport end value| Current Viewport value | Current Viewport AnimationCompleteOnUpdate | Viewport update animation requested | Viewport update value | Result |
+| ----- | --- | --------- | ------------- | ------------------------- | ------------------- | ----- | ------ |
+| YES   | VX1 | VX2       |  VXt          |   **TRUE**                    | YES                 |  VY   | New animation starts between **VX2** and VY |
+| YES   | VX1 | VX2       |  VXt          |   **NO**                    | YES                 |  VY   | New animation starts between **VXt** and VY |
+| NO   | VX | VX       |  VX2          |   **TRUE/FALSE**               | YES                 |  VY   | New animation starts between **VX** and VY |
+| YES   | VX1 | VX2       |  VXt          |   **TRUE/FALSE**               | NO                 |  VY   | Stop current animation and *jump* from VXt to VY |
+
+6. **onAnimationStop** {Function, optional} : This callback will be fired when requested animation is stopped. The object ({animationStep}) will be passed as an argument, which can be used to whether animation was interrupted or completed. This prop can be used to generate continuous animations, that loop animating between set of viewports.
```

Comment:
Should we add `onAnimationStart`, `onAnimationUdpate` for completeness?

Reply:
`onAnimationStop` has a very good use case to kick of new animation right after current one to have continuous loop like animations, but I can't think of any for `onAnimationStart` and `onAnimationUpdate`. May be we can add them in next update and focus on testing current features in an application?